support/releaseinfo/registryclient/client.go

```diff
@@ -223,36 +232,115 @@ func ExtractImageFilesToDir(ctx context.Context, imageRef string, pullSecret []b
 }
 
 func getMetadata(ctx context.Context, imageRef string, pullSecret []byte) ([]distribution.Descriptor, distribution.BlobStore, error) {
+	repo, ref, err := GetRepoSetup(ctx, imageRef, pullSecret)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to repo setup: %w", err)
+	}
+	firstManifest, location, err := manifest.FirstManifest(ctx, ref, repo)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to obtain root manifest for %s: %w", imageRef, err)
+	}
+	_, layers, err := manifest.ManifestToImageConfig(ctx, firstManifest, repo.Blobs(ctx), location)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to obtain image layers for %s: %w", imageRef, err)
+	}
+	return layers, repo.Blobs(ctx), nil
+}
+
+// GetRepoSetup connects to a repo and pulls the imageRef's docker image information from the repo. Returns the repo and the docker image.
+func GetRepoSetup(ctx context.Context, imageRef string, pullSecret []byte) (distribution.Repository, reference.DockerImageReference, error) {
```

Comment:
I'd rather return a pointer here instead of reference.DockerImageReference{} so it's easier for consumers to check

Reply:
Ok I updated that to be a pointer instead.